x-pack/plugins/maps/server/routes.js

```diff
@@ -47,32 +46,70 @@ export function initRoutes(server, licenseUid) {
     }
   });
 
+
+  server.route({
+    method: 'GET',
+    path: `${ROOT}/${EMS_DATA_TMS_PATH}`,
+    handler: async (request, h) => {
+
+      if (!request.query.id ||
+        typeof parseInt(request.query.x, 10) !== 'number' ||
+        typeof parseInt(request.query.y, 10) !== 'number' ||
+        typeof parseInt(request.query.z, 10) !== 'number'
+      ) {
+        server.log('warning', 'Must supply id/x/y/z parameters to retrieve EMS tile');
+        return null;
+      }
+
+      const ems = await getEMSResources(emsClient, mapConfig.includeElasticMapsService, licenseUid, true);
+      const tmsService = ems.tmsServices.find(layer => layer.id === request.query.id);
+      if (!tmsService) {
+        return null;
+      }
+
+      const url = tmsService.url
+        .replace('{x}', request.query.x)
+        .replace('{y}', request.query.y)
+        .replace('{z}', request.query.z);
+
+      const tile = await fetch(url);
+      const arrayBuffer = await tile.arrayBuffer();
+      const buffer = Buffer.from(arrayBuffer);
+      let  response =  h.response(buffer);
+      response = response.bytes(buffer.length);
+      response = response.header('Content-Disposition', 'inline');
+      response = response.header('Content-type', 'image/png');
+      response = response.encoding('binary');
+      return response;
+    }
+  });
+
   server.route({
     method: 'GET',
-    path: `${ROOT}/meta`,
+    path: `${ROOT}/${EMS_META_PATH}`,
     handler: async () => {
 
       let ems;
-      try {
-        ems = await getEMSResources(licenseUid);
-      } catch (e) {
-        server.log('warning', `Cannot connect to EMS, error: ${e}`);
-        ems = {
-          fileLayers: [],
-          tmsServices: []
-        };
+      const emptyResponse = {
+        fileLayers: [],
+        tmsServices: []
+      };
+
+      if (!mapConfig.proxyElasticMapsServiceInMaps) {
```

Comment:
I don't think you need any logic for checking `proxyElasticMapsServiceInMaps`. The client will only call the `meta` endpoint if proxyElasticMapsServiceInMaps is true and the server should fetch EMS meta data

Reply:
right, this call was removed from the client.